### PR TITLE
Documentation/improve experimental ray serve advanced guide

### LINKED ISSUE #45026: [Ray Serve] Enhance and bug fix to experimental multi application in docker container guide
### Description

While playing around with this experimental feature, I had few suggestions on improving the documentation.

1. Bug in `resnet.Dockerfile`  -> `RUN torch==2.0.1 torchvision==0.15.2` should be `RUN pip install torch==2.0.1 torchvision==0.15.2`
2. Using `podman` to build images instead of docker to avoid the issues of dealing with podman and docker daemon.
3. Bug fix in Dockerfile to include `PYTHONPATH` environment variable as without it I faced `ModuleNotFoundError: No module named 'resnet50_example' and `ModuleNotFoundError: No module named 'whisper_example' 

### Link

https://docs.ray.io/en/latest/serve/advanced-guides/multi-app-container.html

### PR DESCRIPTION
**Pull request by dudeperf3ct**
## Why are these changes needed?

Update the experimental feature guide on multi-container deployment approach for Ray Serve.

## Related issue number

Closes: #45026

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
@zcin PTAL

File: doc/source/serve/advanced-guides/multi-app-container.md [hunk 1]
```diff
@@ -57,29 +59,31 @@ RUN sudo apt-get update && sudo apt-get install curl -y
...
 ::::
 
-Then, build the corresponding Docker images and push it to your choice of Docker registry. This tutorial uses `alice/whisper_image:latest` and `alice/resnet_image:latest` as placeholder names for the images, but make sure to swap out `alice` for a repo name of your choice.
+Then, build the corresponding Docker images and push it to your choice of Docker registry using `podman`. This tutorial uses `alice/whisper_image:latest` and `alice/resnet_image:latest` as placeholder names for the images, but make sure to swap out `alice` for a repo name of your choice.
 
 ::::{tab-set}
 :::{tab-item} Whisper
 ```bash
-# Build the Docker image from the Dockerfile
+# Build the Podman image from the Dockerfile using podman
```
Comment (aslonnie) [inline doc/source/serve/advanced-guides/multi-app-container.md:73]: maybe just say build the "container image"? I am not sure if "podman" image is a common saying.

File: doc/source/serve/advanced-guides/multi-app-container.md [hunk 2]
```diff
@@ -57,29 +59,31 @@ RUN sudo apt-get update && sudo apt-get install curl -y
...
 # Download the source code for the ResNet application into `resnet50_example.py`.
 RUN curl -O https://raw.githubusercontent.com/ray-project/ray/master/doc/source/serve/doc_code/resnet50_example.py
+
+ENV PYTHONPATH "${PYTHONPATH}:/home/ray"
 ```
 :::
 ::::
 
-Then, build the corresponding Docker images and push it to your choice of Docker registry. This tutorial uses `alice/whisper_image:latest` and `alice/resnet_image:latest` as placeholder names for the images, but make sure to swap out `alice` for a repo name of your choice.
+Then, build the corresponding Docker images and push it to your choice of Docker registry using `podman`. This tutorial uses `alice/whisper_image:latest` and `alice/resnet_image:latest` as placeholder names for the images, but make sure to swap out `alice` for a repo name of your choice.
```
Comment (aslonnie) [inline doc/source/serve/advanced-guides/multi-app-container.md:68]: you might want to replace `Docker images` `Docker registry` into "container images", "container registry`. and maybe remove the "using podman". when pushing an image to a registry, if it is using `podman` is not really important. there are many other tools in the field that can push images.

**Review by aslonnie — COMMENTED** [review]
will leave @zcin as main reviewer, let me rereview if needs docs approval or needs merging.

File: doc/source/serve/advanced-guides/multi-app-container.md [hunk 3]
```diff
@@ -57,39 +59,41 @@ RUN sudo apt-get update && sudo apt-get install curl -y
 
 # Download the source code for the ResNet application into `resnet50_example.py`.
 RUN curl -O https://raw.githubusercontent.com/ray-project/ray/master/doc/source/serve/doc_code/resnet50_example.py
+
+ENV PYTHONPATH "${PYTHONPATH}:/home/ray"
```
Comment (zcin) [inline doc/source/serve/advanced-guides/multi-app-container.md:63]: Could you add a comment explaining why this is necessary, for anyone who runs into this in the future?

**Review by zcin — APPROVED** [review]
Looks good, thanks for contributing these improvements!

**Pull request merged**